intuita-webview/src/main/App.tsx

```diff
@@ -65,19 +66,44 @@ function App() {
 		};
 	}, []);
 
+	const handlePanelTabClick = (id: TabKind) => {
+		vscode.postMessage({
+			kind: 'webview.main.setState',
+			activeTabId: id,
+		});
+	};
+
 	return (
 		<main className="App" ref={ref}>
 			<VSCodePanels
 				activeid={activeTabId}
 				className="h-full w-full vscode-panels"
 			>
-				<VSCodePanelTab className="vscode-tab" id={TabKind.codemods}>
+				<VSCodePanelTab
+					className="vscode-tab"
+					id={TabKind.codemods}
+					onClick={() => {
+						handlePanelTabClick(TabKind.codemods);
+					}}
+				>
 					Codemods Discovery
 				</VSCodePanelTab>
-				<VSCodePanelTab className="vscode-tab" id={TabKind.codemodRuns}>
+				<VSCodePanelTab
+					className="vscode-tab"
+					id={TabKind.codemodRuns}
+					onClick={() => {
```

Comment:
can you check if we  can use `activeidChanged` callback on `VSCodePanels` component instead of adding click handlers to each tab?